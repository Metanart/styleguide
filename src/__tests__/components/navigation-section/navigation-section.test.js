import React from 'react';
import renderer, { create, act } from 'react-test-renderer';
import NavigationSection from '../../../components/navigation-section/navigation-section';
import navigationSectionProps from './navigation-section.mockup.json';

describe('Navigation-Section Tests:', () => {
    it('When we open app', () => {
        const props = navigationSectionProps.sections[0];
        const component = renderer.create(<NavigationSection {...props} />);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('When we toggle section to open state', () => {
        const props = navigationSectionProps.sections[0];
        const component = renderer.create(<NavigationSection {...props} />);

        const instance = component.getInstance();

        expect(instance.state.isOpened).toBe(false);
        instance.toggleSection();

        expect(instance.state.isOpened).toBe(true);
    });

    it('When we toggle other section', () => {
        const props = navigationSectionProps.sections[1];
        let component = renderer.create(<NavigationSection {...props} />);

        const instance = component.getInstance();

        expect(instance.state.isOpened).toBe(false);

        const updatedProps = Object.assign({}, props, {
            isOpened: true,
        });

        component = renderer.create(<NavigationSection {...updatedProps} />);

        instance.toggleSection();

        expect(instance.state.isOpened).toBe(true);
    });

    it('When we toggle isOpened for section', () => {
        let root;

        const props = navigationSectionProps.sections[1];

        act(() => {
            root = create(<NavigationSection {...props} />);
        });

        expect(root.toJSON()).toMatchSnapshot();

        act(() => {
            const updatedProps = Object.assign({}, props, {
                isOpened: true,
            });
            root.update(<NavigationSection {...updatedProps} />);
        });

        expect(root.toJSON()).toMatchSnapshot();
    });
});
