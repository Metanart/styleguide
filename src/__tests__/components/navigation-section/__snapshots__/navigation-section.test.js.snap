// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation-Section Tests: When we open app 1`] = `
<div
  className="sc-AxjAm bkRBQT"
>
  <div
    className="sc-AxirZ kngXYT"
    onClick={[Function]}
  >
    Structure
  </div>
  <div
    className="sc-AxiKw fXmGGd"
  >
    <div
      className="sc-AxhCb eWEqxl"
    >
      <a
        className="sc-AxhUy fGlHFc"
        data-vrt-locator="link"
        href="#Structure-Component"
      >
        Component
      </a>
    </div>
  </div>
</div>
`;

exports[`Navigation-Section Tests: When we toggle isOpened for section 1`] = `
<div
  className="sc-AxjAm bkRBQT"
>
  <div
    className="sc-AxirZ kngXYT"
    onClick={[Function]}
  >
    Components
  </div>
  <div
    className="sc-AxiKw fXmGGd"
  >
    <div
      className="sc-AxhCb eWEqxl"
    >
      <a
        className="sc-AxhUy fGlHFc"
        data-vrt-locator="link"
        href="#Components-ErrorBoundary"
      >
        ErrorBoundary
      </a>
    </div>
  </div>
</div>
`;

exports[`Navigation-Section Tests: When we toggle isOpened for section 2`] = `
<div
  className="sc-AxjAm bkRBQT"
>
  <div
    className="sc-AxirZ kngXYT"
    onClick={[Function]}
  >
    Components
  </div>
  <div
    className="sc-AxiKw fVTAYb"
  >
    <div
      className="sc-AxhCb eWEqxl"
    >
      <a
        className="sc-AxhUy fGlHFc"
        data-vrt-locator="link"
        href="#Components-ErrorBoundary"
      >
        ErrorBoundary
      </a>
    </div>
  </div>
</div>
`;
